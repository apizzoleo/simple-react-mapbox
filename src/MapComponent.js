import React,{Component} from 'react';
import ReactMapGL from 'react-map-gl';

export default class Map extends Component {

  state = {
    mapboxApiAccessToken: 'pk.eyJ1IjoiYXBpenpvbGVvIiwiYSI6ImNqbWtkd240eTBhOXQza21taTJhcW9hbWgifQ.uMey3ZGAytPjd4x-94Osxg',
    viewport: {
      width: 600,
      height: 400,
      latitude: 41.8919300,
      longitude: 12.5113300,
      zoom: 8
    }
  };

  render() {
    const { mapboxApiAccessToken } = this.state;
    return (
      <ReactMapGL
        mapboxApiAccessToken={mapboxApiAccessToken}
        {...this.state.viewport}
        onViewportChange={(viewport) => this.setState({viewport})}
      />
    );
  }
}