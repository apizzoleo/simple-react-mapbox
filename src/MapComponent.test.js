import React from "react";
import MapComponent from "./MapComponent";
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

test('MapComponent renderer', () => {
  const component = renderer.create(
    <MapComponent></MapComponent>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('MapComponent mount', () => {
  const wrapper = mount(<MapComponent />);
});
