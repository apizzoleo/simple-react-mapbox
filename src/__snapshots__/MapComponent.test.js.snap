// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapComponent renderer 1`] = `
<div
  id="map"
>
  <div
    style={
      Object {
        "cursor": "grab",
        "height": "100%",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "position": "absolute",
            "visibility": "inherit",
            "width": "100%",
          }
        }
      />
      <div
        style={
          Object {
            "height": 0,
            "overflow": "visible",
            "width": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;
